api/v1beta1/nodemaintenance_webhook.go

```diff
@@ -160,6 +165,25 @@ func (v *NodeMaintenanceValidator) validateNoNodeMaintenanceExists(nodeName stri
 	return nil
 }
 
+func (v *NodeMaintenanceValidator) validateNodeIsReady(nodeName string) error {
+	node, err := getNode(nodeName, v.client)
+	if err != nil {
+		return fmt.Errorf("could not get node for validating node is Ready, please try again: %v", err)
+	} else if node == nil {
+		return fmt.Errorf(ErrorNodeNotExists, nodeName)
+	}
+	// find the status of node condition Ready
+	nodeConditionByType := make(map[corev1.NodeConditionType]corev1.NodeCondition)
+	for _, nc := range node.Status.Conditions {
+		nodeConditionByType[nc.Type] = nc
+	}
+	nodeConditionReady := nodeConditionByType[corev1.NodeReady].Status
```

Comment:
this will panic when there is no `Ready` condition (which can happen on new nodes AFAIK)

Reply:
I am not aware of this scenario, but adding a check on whether the condition exists is a good point.